### PR TITLE
LPS-143724 Notify unsaved changes when navigating away from edit blueprint form

### PR DESCRIPTION
**Pull request by oliv-yu**
https://issues.liferay.com/browse/LPS-143724 - When editing a Blueprint, I want to be notified when there are unsaved changes before closing

- Hook `useShouldConfirmBeforeNavigate` was created in order to add into any form that needs confirmation upon navigating away
- Added to edit element too! It is based on whether the initial and final string values differ.
- It seemed like blueprint title and description would actually not trigger the prompt if they were edited, so I tucked them into values of formik. It seemed a bit confusing the way it originally worked, assigning the title/description into a new object with defaultValue and then using form inputs that get read and added to the submission object. So hopefully this update makes it easier to follow.

Closing browser (not customizable, depends on browser):
Screen Shot 2022-02-28 at 3 13 40 PM

Clicking cancel / back button:
Screen Shot 2022-02-28 at 3 13 16 PM

### CODE REVIEW

File: modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/utils/useShouldConfirmBeforeNavigate.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import {useEffect} from 'react';
+
+/**
+ * This ensures that a confirmation appears upon navigating away if there are
+ * warnings such as unsaved changes. It adds a listener to the window to
+ * detect when the user closes the window or refreshes the page. There is also
+ * an SPA lifecycle event used to detect when the user cancels or navigates to
+ * a new page within Liferay.
+ *
+ *
```
[the remark this answers is not in the document]
Comment (oliv-yu) [reply modules/dxp/apps/search-experiences/search-experiences-web/src/main/resources/META-INF/resources/sxp_blueprint_admin/js/utils/useShouldConfirmBeforeNavigate.js]: Done!

**Comment by joshchong** [comment]
ci:test:bundle

**Comment by joshchong** [comment]
ci:test:search

**Comment by joshchong** [comment]
continued in https://github.com/liferay-search/liferay-portal/pull/314

**Pull request closed without merging**